detection_rules/rule.py

```diff
@@ -1033,6 +1067,10 @@ def version_lock(self):
     def set_version_lock(self, value):
         from .version_lock import VersionLock
 
+        err_msg = "Cannot set the version lock when the versioning strategy is configured to bypass the version lock." \
+                  " Set `bypass_version_lock` to `false` in the rules config to use the version lock."
+        assert not RULES_CONFIG.bypass_version_lock, err_msg
```

Comment:
nice - I think we should also raise an error within `TOMLRuleContents.version_lock` as well if `BYPASS_VERSION_LOCK` is `True`.